pkg/webui/console/store/middleware/logics/gateways.js

```diff
@@ -49,7 +49,19 @@ const getGatewayLogic = createRequestLogic({
     const gtw = await tts.Gateways.getById(id, selector)
     dispatch(gateways.startGatewayEventsStream(id))
 
-    return gtw
+    // Check if gateway is managed by TTI
+    let managed = undefined
+    try {
+      managed = await tts.Gateways.getManagedGateway(id, [
+        'wifi_profile_id',
+        'ethernet_profile_id',
+        'version_ids',
+        'wifi_mac_address',
+        'ethernet_mac_address',
+      ])
+    } catch (e) {}
```

Comment:
What do you recommend for this? With these request we check if the gateway is managed. If the request is successful that means that it is managed otherwise it is not.

Reply:
Make sure to only ignore the expected errors and rethrow any unexpected ones. `lib/errors/utils.js` has a bunch of functions that can identify an error.